modules-js/hapi-next/src/hapi-next.ts

```diff
@@ -82,22 +100,49 @@ export function makeRoutesForNextApp(
   return routes;
 }
 
-export function makeNextHandler(app: next.Server) {
-  const requestHandler = app.getRequestHandler();
-
-  return async (request, h) => {
+/**
+ * Creates a Hapi handler method to generate HTML from Next. If the Hapi route
+ * has any path parameters, those are merged into the query parameters and
+ * passed to Next.
+ *
+ * @param app Next app
+ * @param page If provided, the Next page that should be rendered. Used if this
+ * is different from the request path, for example when we’re using a REST-like
+ * path. E.g.: /certificates/12345 should be handled by the "certificates" page.
+ */
+export function makeNextHandler(
+  app: next.Server,
+  page: string | null = null
+): Lifecycle.Method {
+  return async (request: HapiRequest, h: ResponseToolkit) => {
     const {
       raw: { req, res },
+      query,
+      path,
+      params,
     } = request;
 
+    // Be consistent about having Next pages not have a slash. Unless it’s only
```

Comment:
Minor, but adds clarity:

`// Be consistent about having Next pages not have a` **`trailing`** `slash. Unless it’s only`